internal/host/statemachine.go

```diff
@@ -769,6 +770,62 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th TransitionHandler) stat
 		},
 	})
 
+	// Time out while host installationInProgress
+	sm.AddTransitionRule(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusInstallingInProgress)},
+		Condition: stateswitch.And(
+			stateswitch.Not(If(SoftTimeoutsEnabled)),
+			IsInStages(models.HostStageWritingImageToDisk),
+			th.HasInstallationInProgressTimedOut,
+			stateswitch.Not(shouldIgnoreInstallationProgressTimeout)),
+		DestinationState: stateswitch.State(models.HostStatusError),
+		PostTransition:   th.PostRefreshHost(statusInfoInstallationInProgressWritingImageToDiskTimedOut),
+		Documentation: stateswitch.TransitionRuleDoc{
+			Name:        "Move to error on timeout if host is in particular installation in progress stages",
+			Description: "TODO: Document this transition rule.",
+		},
+	})
+
+	// Time out while host installationInProgress
+	sm.AddTransitionRule(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusInstallingInProgress)},
+		Condition: stateswitch.And(
+			If(SoftTimeoutsEnabled),
+			stateswitch.Not(If(HostStageTimedOut)),
+			stateswitch.Not(IsInStages(models.HostStageWritingImageToDisk, models.HostStageRebooting)),
+			th.HasInstallationInProgressTimedOut,
+			stateswitch.Not(shouldIgnoreInstallationProgressTimeout)),
+		DestinationState: stateswitch.State(models.HostStatusInstallingInProgress),
+		PostTransition:   th.PostHostStageTimeout(statusInfoInstallationInProgressSoftTimedOut),
+		Documentation: stateswitch.TransitionRuleDoc{
+			Name:        "Indicate that timeout occurred and continue installation in particular installation in progress stages",
+			Description: "TODO: Document this transition rule.",
+		},
+	})
+
+	// Time out while host installationInProgress
+	sm.AddTransitionRule(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusInstallingInProgress)},
+		Condition: stateswitch.And(
+			If(SoftTimeoutsEnabled),
+			stateswitch.Not(If(HostStageTimedOut)),
+			IsInStages(models.HostStageWritingImageToDisk),
```

Comment:
What about rebooting?

Looks like  the case for enabled soft timeouts and rebooting isn't here? Maybe I'm missing something.
It would help if you wrote out the intended state machine changes in the PR description or comments.

Reply:
It cannot be rebooting and writing image to disk at the same time

Reply:
>timeout on rebooting is not failing the installation even today, but moves the host to installing-pending-user-action. I don't think we should change it.

This is what I was asking about. Thanks for the explanation.